gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/BaseGitCoreBranch.java

```diff
@@ -74,12 +74,19 @@ public synchronized GitCoreCommit getPointedCommit() throws GitCoreException {
   @Override
   public List<IGitCoreCommit> deriveCommitsUntil(IGitCoreCommit upToCommit) throws GitCoreException {
     LOG.debug(() -> "Entering: branch = '${getFullName()}', upToCommit = ${upToCommit}");
+    @Unique RevWalk walk = getTopoRevWalkFromPointedCommit();
+    try {
+      walk.markUninteresting(walk.parseCommit(repo.gitCoreCommitToObjectId(upToCommit)));
+    } catch (IOException e) {
+      throw new GitCoreException(e);
+    }
+    walk.sort(RevSort.BOUNDARY);
 
     LOG.debug("Starting revwalk");
-    return Iterator.ofAll(getTopoRevWalkFromPointedCommit())
+    return Iterator.ofAll(walk.iterator())
         .takeUntil(revCommit -> revCommit.getId().getName().equals(upToCommit.getHash().getHashString()))
         .map(revCommit -> {
-          LOG.debug(() -> revCommit.getId().getName());
+          LOG.debug(() -> "* " + revCommit.getId().getName());
```

Comment:
string interpolations here?

Reply:
IMO more readable without here :) string interpolations IMO make less sense when there's just one value to be interpolated at the beginning or end of a short string